pkg/gen/dotnet-templates/kind.cs.mustache

```diff
@@ -0,0 +1,68 @@
+// *** WARNING: this file was generated by the Pulumi Kubernetes codegen tool. ***
+// *** Do not edit by hand unless you're certain you know what you are doing! ***
+
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Pulumi.Serialization;
+
+namespace Pulumi.Kubernetes.{{Group}}.{{Version}} {
+    /// <summary>
+    /// {{{Comment}}}{{{PulumiComment}}}
+    /// </summary>
+    public partial class {{Kind}} : Pulumi.CustomResource {
+        {{#Properties}}
+        {{{Comment}}}
+        [Output("{{Name}}")]
+        public {{{PropType}}} {{LanguageName}} { get; private set; } = null!;
+
+        {{/Properties}}
+
+        /// <summary>
+        /// Create a {{Kind}} resource with the given unique name, arguments, and options.
+        /// </summary>
+        ///
+        /// <param name="name">The unique name of the resource</param>
+        /// <param name="args">The arguments used to populate this resource's properties</param>
+        /// <param name="options">A bag of options that control this resource's behavior</param>
+        public {{Kind}}(string name, Types.Inputs.{{Group}}.{{Version}}.{{Kind}}Args? args = null, CustomResourceOptions? options = null)
+            : base("kubernetes:{{URNAPIVersion}}:{{Kind}}", name, SetAPIKindAndVersion(args), MakeResourceOptions(options, ""))
+        {
+        }
+
+        private static ResourceArgs? SetAPIKindAndVersion(Types.Inputs.{{Group}}.{{Version}}.{{Kind}}Args? args) {
```

Comment:
I believe we should never send `null` to the base class arguments. Use `Empty` for the default case.

Reply:
Should we fix the base class constructor to not accept null?

Reply:
It looks like I'm wrong and it's only `Resource` that doesn't accept them https://github.com/pulumi/pulumi/blob/394c91d7f6ab7a4096f4454827690a460f665433/sdk/dotnet/Pulumi/Resources/CustomResource.cs#L34